library/rmc/src/invariant.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+/// This module introduces the Invariant trait and as well as implementation for commonly used types.
```

Comment:
Are there any standard types missing here?

Reply:
From the list of primitive types [here](https://doc.rust-lang.org/std/index.html#primitives), the types missing are:

- never
- fn
- pointer
- reference
- slice
- str
- tuple
- unit

Except for tuple, I don't see a use case for invariant. The tuple is a bit more painful to generate since rust doesn't have variadic templates.

In the next revision, I will include array which I was missing.

The standard library have a lot of other types that we are not implementing yet. Things with pointers and references are a bit trickier to implement. I think having a builder pattern for them will make more sense.